### PR TITLE
Added O365 fields and mapping

### PR DESCRIPTION
**Pull request by teizenman**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/22141

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [X] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

### CODE REVIEW

File: Packs/FeedOffice365/Integrations/FeedOffice365/FeedOffice365.py [hunk 1]
```diff
@@ -169,7 +150,11 @@ def fetch_indicators(client: Client, indicator_type_lower: str, limit: int = -1)
                     "rawJSON": raw_data,
                     "fields": {
                         "port": argToList(item.get('tcpPorts', '')),
-                        "subfeed": item.get('serviceArea', '')
+                        "subfeed": item.get('serviceArea', ''),
```
Comment (avidan-H) [inline Packs/FeedOffice365/Integrations/FeedOffice365/FeedOffice365.py]: Aren't you changing `subfeed` to `service` in your other PR? Worthwhile to preemptively change it here even if that PR hasn't been merged yet.

File: Packs/FeedOffice365/Integrations/FeedOffice365/FeedOffice365.py [hunk 2]
```diff
@@ -169,7 +150,11 @@ def fetch_indicators(client: Client, indicator_type_lower: str, limit: int = -1)
...
+                        "office365expressroute": item.get('expressRoute', ''),
```
Comment (avidan-H) [inline Packs/FeedOffice365/Integrations/FeedOffice365/FeedOffice365.py]: We should only set the boolean values if they are returned, so don't set a default, and don't map to the indicator field _unless_ a non-`None` boolean value is returned from the `.get()`.

File: Packs/FeedOffice365/Integrations/FeedOffice365/FeedOffice365.py [hunk 3]
```diff
@@ -169,7 +150,11 @@ def fetch_indicators(client: Client, indicator_type_lower: str, limit: int = -1)
...
+                        "office365category": item.get('category', ''),
+                        "office365required": item.get('required', ''),
```
Comment (avidan-H) [inline Packs/FeedOffice365/Integrations/FeedOffice365/FeedOffice365.py]: Same here.

**Comment by teizenman** [comment]
@avidan-H all were fixed

File: Packs/FeedOffice365/Integrations/FeedOffice365/FeedOffice365.py [hunk 4]
```diff
@@ -159,22 +155,34 @@ def fetch_indicators(client: Client, indicator_type_lower: str, limit: int = -1)
...
                 for key, val in item.items():
                     if key not in ['ips', 'urls']:
                         raw_data.update({key: val})
+
+                indicator_mapping_fields = {
+                    "port": argToList(item.get('tcpPorts', '')),
+                    "service": item.get('serviceArea', '')
+                }
+
+                if item.get('expressRoute', None):
```
Comment (avidan-H) [inline Packs/FeedOffice365/Integrations/FeedOffice365/FeedOffice365.py]:
```suggestion
                if item.get('expressRoute'):
```
The default value of `.get()` is `None` - no need to explicitly set it to `None`. The same goes for wherever else you explicitly set the default value to `None`.

**Pull request merged**